src/proto/h2/client.rs

```diff
@@ -245,10 +245,11 @@ pin_project! {
         T: Unpin,
     {
         #[pin]
-        select: Select<ConnMapErr<T, B>, StreamFuture<Receiver<Never>>>,
```

Comment:
What speaks against `select: Select<Fuse<ConnMapErr<T, B>>, Fuse<StreamFuture<Receiver<Never>>>>,

Reply:
Uhm, not sure without further context, but I think this won't be enough and the actual fix has to be done in the `poll()` code.

Right now, if the `ConnTask->Select->ConnMap` side returns `Ready` first, it is theoretically possible for a subsequent/spurious poll to go through `ConnTask->Select` again, which would then immediately panic.

This PR rewrites the `ConnTask::poll()` logic (to avoid such execution flow) and also drops the stored `Select` (to avoid part of the issue, but also because it doesn't really seems useful in this context).

Reply:
Oh yes, you are right, the select is panicing not the futures in the select.